$NetBSD: patch-Makefile,v 1.8 2022/02/21 11:47:41 fcambus Exp $

- Honour CFLAGS and CXXFLAGS
- Do not hard-code -pthread, as it is handled by the pkgsrc infrastructure.
- Do not hard-code -ldl

--- Makefile.orig	2022-02-20 07:40:17.000000000 +0000
+++ Makefile
@@ -42,14 +42,14 @@ endif
 
 # If you want to compile mold for debugging, invoke make as
 # `make CXXFLAGS=-g`.
-CFLAGS = -O2
-CXXFLAGS = -O2
+CFLAGS ?= -O2
+CXXFLAGS ?= -O2
 
 MOLD_CXXFLAGS := -std=c++20 -fno-exceptions -fno-unwind-tables \
                  -fno-asynchronous-unwind-tables -Ithird-party/xxhash \
                  -DMOLD_VERSION=\"$(VERSION)\" -DLIBDIR="\"$(LIBDIR)\""
 
-MOLD_LDFLAGS := -pthread -lz -lm -ldl
+MOLD_LDFLAGS := -lz -lm
 
 GIT_HASH := $(shell [ -d .git ] && git rev-parse HEAD)
 ifneq ($(GIT_HASH),)
